datafusion/core/src/execution/session_state.rs

```diff
@@ -943,6 +943,31 @@ impl SessionState {
         let udtf = self.table_functions.remove(name);
         Ok(udtf.map(|x| x.function().clone()))
     }
+
+    fn build_sql_query_planner<'a, S>(&self, provider: &'a S) -> SqlToRel<'a, S>
+    where
+        S: ContextProvider,
+    {
+        let query = SqlToRel::new_with_options(provider, self.get_parser_options());
+
+        // register crate of array expressions (if enabled)
+        #[cfg(feature = "array_expressions")]
+        {
+            let array_planner =
+                Arc::new(ArrayFunctionPlanner::try_new(provider).unwrap()) as _;
+
+            let field_access_planner =
+                Arc::new(FieldAccessPlanner::try_new(provider).unwrap()) as _;
```

Comment:
`get_field` for struct does not need array expression, but I'm not sure should we move it out